Source/DafnyCore/Backends/Dafny/AST.dfy

```diff
@@ -19,13 +38,22 @@ module {:extern "DAST"} DAST {
     Set(element: Type) |
     Multiset(element: Type) |
     Map(key: Type, value: Type) |
+    SetBuilder(element: Type) |
+    MapBuilder(key: Type, value: Type) |
     Arrow(args: seq<Type>, result: Type) |
     Primitive(Primitive) | Passthrough(string) |
     TypeArg(Ident)
 
   datatype Primitive = Int | Real | String | Bool | Char
 
-  datatype ResolvedType = Datatype(path: seq<Ident>) | Trait(path: seq<Ident>) | Newtype(Type)
+  datatype NewtypeRange =
+    | U8 | I8 | U16 | I16 | U32 | I32 | U64 | I64 | U128 | I128 | BigInt
+    | NoRange
```

Comment:
What's the difference between `BigInt` and `NoRange`?

Reply:
More newtypes are coming which are not just integers, like sequences. They will get NoRange instead of BigInt until we might find other categories for them.